src/main/kotlin/io/github/mojira/arisa/ModuleExecutor.kt

```diff
@@ -0,0 +1,304 @@
+import io.github.mojira.arisa.modules.AttachmentModule
+import io.github.mojira.arisa.modules.AttachmentModuleRequest
+import io.github.mojira.arisa.modules.CHKModule
+import io.github.mojira.arisa.modules.CHKModuleRequest
+import io.github.mojira.arisa.modules.CrashModule
+import io.github.mojira.arisa.modules.CrashModuleRequest
+import io.github.mojira.arisa.modules.EmptyModule
+import io.github.mojira.arisa.modules.EmptyModuleRequest
+import io.github.mojira.arisa.modules.FailedModuleResponse
+import io.github.mojira.arisa.modules.FutureVersionModule
+import io.github.mojira.arisa.modules.FutureVersionModuleRequest
+import io.github.mojira.arisa.modules.KeepPrivateModule
+import io.github.mojira.arisa.modules.KeepPrivateModuleRequest
+import io.github.mojira.arisa.modules.ModuleError
+import io.github.mojira.arisa.modules.ModuleResponse
+import io.github.mojira.arisa.modules.OperationNotNeededModuleResponse
+import io.github.mojira.arisa.modules.PiracyModule
+import io.github.mojira.arisa.modules.PiracyModuleRequest
+import io.github.mojira.arisa.modules.RemoveNonStaffMeqsModule
+import io.github.mojira.arisa.modules.RemoveNonStaffMeqsModuleRequest
+import io.github.mojira.arisa.modules.RemoveTriagedMeqsModule
+import io.github.mojira.arisa.modules.RemoveTriagedMeqsModuleRequest
+import io.github.mojira.arisa.modules.ReopenAwaitingModule
+import io.github.mojira.arisa.modules.ReopenAwaitingModuleRequest
+import io.github.mojira.arisa.modules.RevokeConfirmationModule
+import io.github.mojira.arisa.modules.RevokeConfirmationModuleRequest
+import net.rcarz.jiraclient.Issue
+import net.rcarz.jiraclient.JiraClient
+import net.sf.json.JSONObject
+import java.text.SimpleDateFormat
+import java.util.Timer
+import kotlin.concurrent.schedule
+
+class ModuleExecutor(
+    private val jiraClient: JiraClient,
+    private val config: Config
+) {
+    private val attachmentModule: AttachmentModule
+    private val chkModule: CHKModule
+    private val crashModule: CrashModule
+    private val emptyModule: EmptyModule
+    private val keepPrivateModule: KeepPrivateModule
+    private val futureVersionModule: FutureVersionModule
+    private val piracyModule: PiracyModule
+    private val removeNonStaffMeqsModule: RemoveNonStaffMeqsModule
+    private val removeTriagedMeqsModule: RemoveTriagedMeqsModule
+    private val reopenAwaitingModule: ReopenAwaitingModule
+    private val revokeConfirmationModule: RevokeConfirmationModule
+
+    private val ticketCache = mutableListOf<String>()
+    private val ticketTimer = Timer("RemoveCachedTicket", true)
+
+    fun execute() {
+        // Cache issues returned by a query to avoid searching the same query for different modules
+        val queryCache = mutableMapOf<String, List<Issue>>()
+        val processedTickets = mutableMapOf<String, Boolean>()
+        val exec = ::executeModule.partially2(queryCache).partially2(processedTickets)
+
+        exec(Arisa.Modules.Attachment) {
```

Comment:
It would be really nice if we can start adding some tests using the fact that we separated this into a separate file.

That means testing the filter logic we have in execute module, testing the ticket caching, that X module was executed if passed

Reply:
Gonna look into that later

Reply:
#78